block-producer/src/block_builder/prover/block_witness.rs

```diff
@@ -261,7 +261,7 @@ impl BlockWitness {
                 .expect("updated accounts number is greater than or equal to the field modulus"),
         );
 
-        StackInputs::new(stack_inputs)
+        StackInputs::new(stack_inputs).map_err(BlockProverError::FailedToConstructStackInputs)
```

Comment:
Under what circumstances could this error happen? Shouldn't this be just `.expect()` since we know for sure we won't construct invalid stack inputs above?